### PR TITLE
storage: Make some functions global in preparation for UI redesign

### PR DESCRIPTION
**Pull request by mvollmer**
- Expose the content row producers, and generalize "is_partition" etc to "options" for the content row producers, so that we can add more overview specific tuning options later.

 - Expose side panel row and action providers, and add "options" for later tuning.

 - Expose top-level content actions, such as "Create logical volume"

This allows us to use those functions from new UI elements, and it is best to make them global as early as possible so that future changes to the actual code in those functions doesn't lead to constant merge conflicts while the new UI is slowly brewing.

### CODE REVIEW

File: pkg/storaged/content-views.jsx [hunk 1]
```diff
@@ -602,20 +602,29 @@ function append_partitions(client, rows, level, block) {
...
-        const item = (
-            <StorageMenuItem key="create"
-                             onlyNarrow
-                             onClick={create_partition}>
-                {_("Create partition")}
-            </StorageMenuItem>);
+        let btn, item, menu;
 
-        const menu = <StorageBarMenu onlyNarrow menuItems={[item]} isKebab />;
+        if (options.unified) {
```
Comment (mvollmer) [inline pkg/storaged/content-views.jsx:607]: This shouldn't be here yet.

File: pkg/storaged/content-views.jsx [hunk 2]
```diff
@@ -627,7 +636,11 @@ function append_partitions(client, rows, level, block) {
 
         rows.push({
             columns: cols,
-            props: { key: "free-space-" + rows.length.toString(), className: "content-level-" + level }
+            props: {
```
Comment (mvollmer) [inline pkg/storaged/content-views.jsx:639]: Shouldn't be here yet.

File: pkg/storaged/content-views.jsx [hunk 3]
```diff
@@ -650,98 +664,109 @@ function append_partitions(client, rows, level, block) {
...
+                        .then(utils.reload_systemd);
+            }
+        },
+        Inits: [
+            init_active_usage_processes(client, usage)
+        ]
+    });
+}
+
+export function block_menu_items(client, block, options) {
```
Comment (mvollmer) [inline pkg/storaged/content-views.jsx:744]: Not yet.

File: pkg/storaged/content-views.jsx [hunk 4]
```diff
@@ -849,6 +874,159 @@ function install_package(name, progress) {
...
+                                    vals.vdo_options.compression, vals.vdo_options.deduplication,
+                                    "auto", { });
+                            });
+                }
+            }
+        }
+    });
+}
+
+export function vgroup_menu_items(client, vgroup, options) {
```
Comment (mvollmer) [inline pkg/storaged/content-views.jsx:1016]: Not yet.

File: pkg/storaged/iscsi-panel.jsx [hunk 5]
```diff
@@ -30,216 +30,220 @@ import { dialog_open, TextInput, PassInput, SelectRow } from "./dialog.jsx";
...
-                detail: session.data.persistent_address + ":" + session.data.persistent_port
-            };
-        }
+export class IscsiPanel extends React.Component {
+    constructor() {
+        super();
+        this.state = { armed: false };
+    }
+
+    render() {
```
Comment (martinpitt) [inline pkg/storaged/iscsi-panel.jsx:236]: This would be an excellent candidate for a functional component.

**Review by martinpitt — APPROVED** [review]
Excellent, let's get the low-risk, but high-bitrot stuff landed ASAP.

File: pkg/storaged/iscsi-panel.jsx (hunk 5 above)
Comment (mvollmer) [reply pkg/storaged/iscsi-panel.jsx:236]: Yep, it's also a excellent candidate for outight removal. :-)

**Pull request merged**